File: src/main/java/br/com/itaipu/grfe/entity/Escala.java
package br.com.itaipu.grfe.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Table(name = "escala")
public class Escala {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(nullable = false)
    private LocalDateTime dataHoraInicio;

    @Column(nullable = false)
    private LocalDateTime dataHoraFim;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "especialidade_id", nullable = false)
    private Especialidades especialidade;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "funcionario_id", nullable = false)
    private Funcionario funcionario;

}

File: GRFE/src/app/models/escala.ts
export interface Escala {
	id?: number;
	dataHoraInicio: string;
	dataHoraFim: string;
	especialidadeId: number;
	especialidadeNome?: string;
	funcionarioId: number;
	funcionarioNome?: string;
}

export type EscalaRequest = Pick<Escala, 'dataHoraInicio' | 'dataHoraFim' | 'especialidadeId' | 'funcionarioId'>;
